sdk/kiosk/src/client/kiosk-client.ts

```diff
@@ -0,0 +1,476 @@
+		let cap = kioskTx.createKioskAndShare(txb);
+		txb.transferObjects([cap], txb.pure(address, 'address'));
+	}
+
+	/**
+	 * Should be called only after `create` is called.
+	 * It shares the kiosk & transfers the cap to the specified address.
+	 */
+	shareAndTransferCap(
+		txb: TransactionBlock,
+		kiosk: TransactionArgument,
+		cap: TransactionArgument,
+		address: string,
+	) {
+		kioskTx.shareKiosk(txb, kiosk);
+		txb.transferObjects([cap], txb.pure(address, 'address'));
+	}
+
+	/**
+	 * Wraps a kiosk transaction that depends on `kioskOwnerCap`.
+	 * @param txb The Transaction Block
+	 * @param cap The cap as returned from `getOwnedKiosks` SDK call
+	 * @param ownerCap The `ownerCap` object as returned from the `getOwnedKiosk` function
+	 * @param callback The function you want to execute with the ownerCap.
+	 */
+	async ownedKioskTx(
+		txb: TransactionBlock,
+		cap: KioskOwnerCap,
+		callback: (kiosk: TransactionArgument, capObject: TransactionArgument) => Promise<void>,
+	): Promise<void> {
+		let [kiosk, capObject, returnPromise] = this.getOwnerCap(txb, cap);
+
+		await callback(kiosk, capObject);
+
+		this.returnOwnerCap(txb, cap, capObject, returnPromise);
+	}
+
+	/**
+	 * Get an addresses's owned kiosks.
+	 * @param address The address for which we want to retrieve the kiosks.
+	 * @returns An Object containing all the `kioskOwnerCap` objects as well as the kioskIds.
+	 */
+	async getOwnedKiosks(address: string): Promise<OwnedKiosks> {
+		let personalPackageId = PERSONAL_KIOSK_RULE_ADDRESS[this.network];
+		return getOwnedKiosks(this.client, address, {
+			personalKioskType: personalPackageId
+				? `${personalPackageId}::personal_kiosk::PersonalKioskCap`
+				: '',
+		});
+	}
+
+	/**
+	 * Fetches the kiosk contents.
+	 * @param kioskId
+	 * @param options
+	 * @returns
+	 */
+	async getKiosk(kioskId: string, options: FetchKioskOptions): Promise<KioskData> {
+		return (
+			await fetchKiosk(
+				this.client,
+				kioskId,
+				{
+					limit: 1000,
+				},
+				options,
+			)
+		).data;
+	}
+
+	/**
+	 * Query the Transfer Policy(ies) for type `T`.
+	 * @param itemType The Type we're querying for (E.g `0xMyAddress::hero::Hero`)
+	 */
+	async getTransferPolicies(itemType: string) {
+		return queryTransferPolicy(this.client, itemType);
+	}
+
+	/**
+	 * A function to purchase and resolve a transfer policy.
+	 * If the transfer policy has the `lock` rule, the item is locked in the kiosk.
+	 * Otherwise, the item is placed in the kiosk.
+	 * @param txb The Transaction Block
+	 * @param itemType The type of the item
+	 * @param itemId The id of the item
+	 * @param price The price of the specified item
+	 * @param kiosk The kiosk the item resides in
+	 * @param ownedKiosk The kiosk the item will be placed or locked into
+	 * @param ownedKioskCap The OwnerCap as retrieved from the `getOwnerCap` function
+	 * @param options Currently has `extraArgs`, which can be used for custom rule resolvers.
+	 */
+	async purchaseAndResolve(
```

Comment:
having this many positional arguments can be a pain to deal with, it might be better to accept an object with these arguments as properties instead.  This probably applies to several methods in the client

Reply:
Totally agreed. I was also thinking about it but wanted to not divert too much from V1, but indeed now is the time to get this right. 👍